python/AI/vector_store.py

```diff
@@ -0,0 +1,66 @@
+# # Copyright (c) 2024 Microsoft Corporation. All rights reserved.
+# # Licensed under the MIT license. See LICENSE file in the project.
+# #
+from typing import Any
+
+import duckdb
+import lancedb
+import pyarrow as pa
+
+from python.helpers.constants import CACHE_PATH
+
+table_missing_msg = "Table not initialized"
+
+
+class VectorStore:
+    table = None
+    duckdb_data = None
+
+    def __init__(
+        self,
+        table_name: str | None = None,
+        path: str = CACHE_PATH,
+        schema: pa.Schema = None,
+        db_index: list[str] | None = None,
+    ):
+        self.db_connection = lancedb.connect(path)
+        self.db_index = db_index
+        if table_name is not None:
+            self.table = self.db_connection.create_table(
+                table_name, schema=schema, exist_ok=True
+            )
+            self.duckdb_data = self.table.to_lance()
+
+        if self.table is not None and db_index is not None:
+            self.table.create_fts_index(db_index, replace=True)
```

Comment:
would fts index be slower than vector index?

Reply:
Actually I'm not using the index at all anymore so I'll remove it for clarity